src/Microsoft.OData.Core/UriParser/Binders/FunctionCallBinder.cs

```diff
@@ -234,7 +234,28 @@ internal QueryNode BindFunctionCall(FunctionCallToken functionCallToken)
 
             // If there isn't, bind as Uri function
             // Bind all arguments
-            List<QueryNode> argumentNodes = new List<QueryNode>(functionCallToken.Arguments.Select(ar => this.bindMethod(ar)));
+            List<QueryNode> argumentNodes = functionCallToken.Arguments.Select(argument =>
+            {
+                // If the function is IsOf or Cast and the argument is a dotted identifier, we need to bind it differently
+                string functionCallTokenName = IsUnboundFunction(functionCallToken.Name);

```

Comment:
IsUnboundFunction returns a string? It looks non-sense.

Reply:
`IsUnboundFunction` gets `functionCallToken.Name` in any case (lower-case, upper case or camel case) and compare with the `isof` and `cast` using `this.state.Configuration.EnableCaseInsensitiveUriFunctionIdentifier` config

[private string IsUnboundFunction(string functionName)](https://github.com/OData/odata.net/blob/main/src/Microsoft.OData.Core/UriParser/Binders/FunctionCallBinder.cs#L635)

[private static readonly string[] UnboundFunctionNames](https://github.com/OData/odata.net/blob/main/src/Microsoft.OData.Core/UriParser/Binders/FunctionCallBinder.cs#L29)